internal/dataplane/dataplane_tpl.go

```diff
@@ -0,0 +1,226 @@
+package dataplane
+
+import (
+	"bytes"
+	"encoding/json"
+	"strconv"
+	"text/template"
+)
+
+type dataplaneConfig struct {
+	consul    consulConfig
+	service   serviceConfig
+	logging   loggingConfig
+	xdsServer xdsServerConfig
+	envoy     envoyConfig
+	telemetry telemetryConfig
+}
+
+type dataplaneConfigTmplArgs struct {
```

Comment:
Instead of using templating, do you think we can rely only on `encoding/json`, possibly with custom `MarshalJSON` method implementations where necessary (for including/excluding fields)?

Reply:
Thanks for pointing it out. Using `json.Marshal` makes it much more easier to generate the JSON. Removed the old behaviour based on templating